api/src/paths/project/{projectId}/survey/{surveyId}/observations/create.ts

```diff
@@ -0,0 +1,143 @@
+import { observationPostResponseObject } from '../../../../../../openapi/schemas/observation';
+import { postBlockObservationSQL } from '../../../../../../queries/observation/observation-create-queries';
+import { getLogger } from '../../../../../../utils/logger';
+import { logRequest } from '../../../../../../utils/path-utils';
+
+const defaultLog = getLogger('paths/project/{projectId}/survey/{surveyId}/observations/create');
+
+export const POST: Operation = [
+  logRequest('paths/project/{projectId}/survey/{surveyId}/observations/create', 'POST'),
+  createObservation()
+];
+
+POST.apiDoc = {
+  description: 'Create a new observation.',
+  tags: ['observation'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  requestBody: {
+    description: 'Observation post request object.',
+    content: {
+      'application/json': {
+        schema: {
+          title: 'Observation request object',
+          type: 'object',
+          required: ['observation_type'],
+          properties: {
+            observation_type: {
+              title: 'type of observation (eg block)',
+              type: 'string'
+            },
+            observation_post_data: {
+              title: 'Generic observation json data',
+              type: 'object',
+              properties: {}
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Observation response object.',
+      content: {
+        'application/json': {
+          schema: {
+            ...(observationPostResponseObject as object)
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Creates a new observation record.
+ *
+ * @returns {RequestHandler}
+ */
+export function createObservation(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug({
+      label: 'createObservation',
+      message: 'params and body',
+      'req.params': req.params,
+      'req.body': req.body
+    });
+
+    if (!req.params.surveyId) {
+      throw new HTTP400('Missing required path param `surveyId`');
+    }
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    let sanitizedObservationData = null;
+
+    if (req.body.observation_type === 'block') {
+      sanitizedObservationData = (req.body && new PostBlockObservationObject(req.body.observation_post_data)) || null;
+    }
+
+    if (!sanitizedObservationData) {
+      throw new HTTP400('Missing observation data');
+    }
+
+    let observationSQLStatement = null;
+
+    if (req.body.observation_type === 'block') {
```

Comment:
so is it possible to not duplicate these identical if statements somehow? like put all the logic into one conditional?

Reply:
yeah .. was talking to Nick earlier.  There is ... but might be good to wait until we know more about the requirements of the other observations before doing that ... just in case this whole approach has to be scrapped

Reply:
ok yeah that sounds good then